storage/google/cloud/storage/blob.py

```diff
@@ -613,8 +640,213 @@ def _check_response_error(request, http_response):
             raise make_exception(faux_response, http_response.content,
                                  error_info=request.url)
 
+    def _get_writable_metadata(self):
+        """Get the object / blob metadata which is writable.
+
+        This is intended to be used when creating a new object / blob.
+
+        See the `API reference`_ for more information, the fields marked as
+        writable are:
+
+        * ``acl``
+        * ``cacheControl``
+        * ``contentDisposition``
+        * ``contentEncoding``
+        * ``contentLanguage``
+        * ``contentType``
+        * ``crc32c``
+        * ``md5Hash``
+        * ``metadata``
+        * ``name``
+        * ``storageClass``
+
+        For now, we don't support ``acl``, access control lists should be
+        managed directly through :class:`ObjectACL` methods.
+
+        .. _API reference: https://cloud.google.com/storage/\
+                           docs/json_api/v1/objects
+        """
+        # NOTE: This assumes `self.name` is unicode.
+        object_metadata = {'name': self.name}
+        for key in self._changes:
+            if key in _WRITABLE_FIELDS:
+                object_metadata[key] = self._properties[key]
+
+        return object_metadata
+
+    def _get_upload_arguments(self, client, content_type):
+        """Get required arguments for performing an upload.
+
+        The content type returned will be determined in order of precedence:
+
+        - The value passed in to this method (if not :data:`None`)
+        - The value stored on the current blob
+        - The default value ('application/octet-stream')
+
+        :type client: :class:`~google.cloud.storage.client.Client`
+        :param client: (Optional) The client to use.  If not passed, falls back
+                       to the ``client`` stored on the blob's bucket.
+
+        :type content_type: str
+        :param content_type: Type of content being uploaded (or :data:`None`).
+
+        :rtype: tuple
+        :returns: A quadruple of
+
+                  * An
+                    :class:`~google.auth.transport.requests.AuthorizedSession`
+                  * A header dictionary
+                  * An object metadata dictionary
+                  * The ``content_type`` as a string (according to precedence)
+        """
+        transport = self._make_transport(client)
```

Comment:
I **can** make that change, but keep in mind this is temporary until the `httplib2` swap is done (which I want to finish ASAP).